packages/bridging/src/BridgingSdk/getQuoteWithBridge.ts

```diff
@@ -1,119 +1,214 @@
-import { getCacheKey } from './helpers'
+import { isHookBridgeProvider, isReceiverAccountBridgeProvider } from '../utils'
+import { getIntermediateSwapResult } from './getIntermediateSwapResult'
 
 export async function getQuoteWithBridge<T extends BridgeQuoteResult>(
-  params: GetQuoteWithBridgeParams<T>,
+  provider: BridgeProvider<T>,
+  params: GetQuoteWithBridgeParams,
 ): Promise<BridgeQuoteAndPost> {
-  const { provider, swapAndBridgeRequest, advancedSettings, tradingSdk, bridgeHookSigner } = params
-  const {
-    kind,
-    sellTokenChainId,
-    sellTokenAddress,
-    buyTokenChainId,
-    buyTokenAddress,
-    amount,
-    signer: signerLike,
-    ...rest
-  } = swapAndBridgeRequest
-
-  const adapter = getGlobalAdapter()
-  const signer = signerLike ? adapter.createSigner(signerLike) : adapter.signer
+  const { kind } = params.swapAndBridgeRequest
 
+  // Ensure the quote request is for a sell order (only type supported for now)
   if (kind !== OrderKind.SELL) {
     throw new Error('Bridging only support SELL orders')
   }
 
-  log(
-    `Cross-chain ${kind} ${amount} ${sellTokenAddress} (source chain ${sellTokenChainId}) for ${buyTokenAddress} (target chain ${buyTokenChainId})`,
-  )
+  // If the provider relies on hooks
+  if (isHookBridgeProvider(provider)) {
+    return getQuoteWithHookBridge(provider, params)
+  }
 
-  // Get the mocked hook (for estimating the additional swap costs)
-  const bridgeRequestWithoutAmount = await getBaseBridgeQuoteRequest({
-    swapAndBridgeRequest: swapAndBridgeRequest,
-    provider,
-    intermediateTokensCache: params.intermediateTokensCache,
-  })
+  // If the provider doesn't rely on hooks
+  if (isReceiverAccountBridgeProvider(provider)) {
+    return getQuoteWithReceiverAccountBridge(provider, params)
+  }
 
-  // Get the hook mock for cost estimation
-  const hookEstimatedGasLimit = await provider.getGasLimitEstimationForHook(bridgeRequestWithoutAmount)
-  const mockedHook = getHookMockForCostEstimation(hookEstimatedGasLimit)
-  log(`Using mocked hook for swap gas estimation: ${JSON.stringify(mockedHook)}`)
-
-  const { sellTokenAddress: intermediateToken, sellTokenDecimals: intermediaryTokenDecimals } =
-    bridgeRequestWithoutAmount
-
-  // Estimate the expected amount of intermediate tokens received in CoW Protocol's swap
-  const swapParams: WithPartialTraderParams<TradeParameters> = {
-    ...rest,
-    kind,
-    chainId: sellTokenChainId,
-    sellToken: sellTokenAddress,
-    buyToken: intermediateToken,
-    buyTokenDecimals: intermediaryTokenDecimals,
-    amount: amount.toString(),
-    signer,
+  throw new Error('Provider type is unknown: ' + provider.type)
+}
+
+// Common setup logic for both bridge provider types
+
+// Common helper function for getting swap quotes
+
+/**
+ * Create a postSwapOrderFromQuote function that can be used to post the swap order from the quote
+ *
+ * @param params
+ * @returns
+ */
+function createPostSwapOrderFromQuote(params: {
```

Comment:
Nitpick: to make the function easier to read, maybe we should define the params type as a separate type/interface